### PR TITLE
Check if filter is an array

### PR DESCRIPTION
**Pull request by orociscaverai**
| Q             | A
| ------------- | ---
| Branch?       | master for features / x.y for bug fixes 
| Bug/Hotfix?   | no
| Refactoring?  | yes
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes

Check if the filter is an array

### CODE REVIEW

File: src/Mado/QueryBundle/Exceptions/InvalidFiltersException.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Mado\QueryBundle\Exceptions;
+
+use Exception;
+use Throwable;
+
+final class InvalidFiltersException extends Exception
+{
+    public function __construct(string $message = "", int $code = 0, Throwable $previous = null)
```
Comment (sensorario) [inline src/Mado/QueryBundle/Exceptions/InvalidFiltersException.php]: Why overwrite constructor? Please remove it!!

File: src/Mado/QueryBundle/Repositories/BaseRepository.php [hunk 2]
```diff
@@ -146,6 +147,13 @@ public function setQueryOptionsFromRequest(Request $request = null)
         return $this;
     }
 
+    private function ensureFilterIsValid($filters )
+    {
+        if (!is_array($filters)) {
+            throw new InvalidFiltersException("Wrong query string exception: " . var_export($filters, true) . "\n Please check query string should be something like this http://127.0.0.1:8000/?filtering[status]=todo");
```
Comment (sensorario) [inline src/Mado/QueryBundle/Repositories/BaseRepository.php]: "Lines SHOULD NOT be longer than 80 characters; lines longer than that SHOULD be split into multiple subsequent lines of no more than 80 characters each.

https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#23-lines

File: src/Mado/QueryBundle/Exceptions/InvalidFiltersException.php [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Mado\QueryBundle\Exceptions;
+
+use Exception;
+use Throwable;
+
+final class InvalidFiltersException extends Exception
```
Comment (sensorario) [inline src/Mado/QueryBundle/Exceptions/InvalidFiltersException.php]: 👍

**Comment by sensorario** [comment]
@AlessandroMinoccheri we need your approval!!!!

**Comment by sensorario** [comment]
@orociscaverai please take care at pull request description table: "master for features / x.y for bug fixes" must be replaced with "2.3".

File: src/Mado/QueryBundle/Repositories/BaseRepository.php (hunk 2 above)
Comment (sensorario) [reply src/Mado/QueryBundle/Repositories/BaseRepository.php]: @orociscaverai up

**Comment by AlessandroMinoccheri** [comment]
Can we merge It?

File: src/Mado/QueryBundle/Repositories/BaseRepository.php (hunk 2 above)
Comment (sensorario) [reply src/Mado/QueryBundle/Repositories/BaseRepository.php]: @orociscaverai up

**Pull request merged**